frontend/src/component/AddVisitCountry.tsx

```diff
@@ -1,43 +1,100 @@
 import {Country} from "../model/Country";
-import {SyntheticEvent} from "react";
-import "./AddVisitCountry.css";
+import {SyntheticEvent, useState} from "react";
 import {Traveller} from "../model/User";
 import {useNavigate} from "react-router-dom";
 import {updateUser} from "../apiCalls";
-import {Autocomplete, TextField} from "@mui/material";
+import {Autocomplete, Box, Button, Chip, LinearProgress, TextField, Typography} from "@mui/material";
+import useCountries from "../hook/useCountries";
+import {AddCircle} from "@mui/icons-material";
 
 type AddVisitCountryProps = {
-    countries: Country[],
     loggedInTraveller: Traveller
 }
-export default function AddVisitCountry(props: AddVisitCountryProps) {
+export default function AddVisitedCountry(props: AddVisitCountryProps) {
     const navigate = useNavigate()
+    const {countries} = useCountries()
+    const [selectedCountries, setSelectedCountries] = useState<Country[]>([])
 
-    function addCountryToUser(event: SyntheticEvent<Element, Event>, value: Country | null) {
+    function addCountriesToUser() {
         if (!props.loggedInTraveller.visitedCountries) {
             props.loggedInTraveller.visitedCountries = []
         }
+        const newPlusOldCountries = [...props.loggedInTraveller.visitedCountries, ...selectedCountries]
+        const copyLoggedInTraveller = props.loggedInTraveller
+        copyLoggedInTraveller.visitedCountries = newPlusOldCountries
+
+        updateUser(props.loggedInTraveller.id, copyLoggedInTraveller)
+            .then(() => {
+                props.loggedInTraveller.visitedCountries = newPlusOldCountries
```

Comment:
Why would you assign the countries like this?
I'm probably blind, but I'm not seeing the purpose since you'd have to use the setter to update the values

Reply:
idk it works like that 💭

Reply:
I find that peculiar

Reply:
We'll chat about it live :coffee: